### PR TITLE
docs: update Ti.Android.R and Ti.App.Android.R docs

### PR DESCRIPTION
**Pull request by drauggres**
In docs before this PR:
1. `R` described as both: `R` property of `Ti.Android` and class `R` (in `Ti.Android` namespase) that extends `Ti.Proxy`
2. Same for `R` in `Ti.App.Android`
3. [Documentation](https://docs.appcelerator.com/platform/latest/#!/api/Titanium.App.Android.R) for `Ti.App.Android.R` was almost empty.

This PR should fix these problems.

Required for correct fix for https://github.com/appcelerator/docs-devkit/issues/29.

### CODE REVIEW

**Review by janvennemann — CHANGES_REQUESTED** [review]
Left a few notes on some phrasing. I would also like keep the type defined as `Titanium.Android.R` instead of renaming it to `RProxy`. See my comment in the the related PR on the docs-devkit how we could handle that.

**Comment by drauggres** [comment]
@janvennemann Thanks for review. 
There are two problems in the generated `.d.ts`:
1. Properties of the `Ti.Android.R` are defined as properties of the instance (not static), so they are not available (e.g. `Ti.Android.R.drawable` will not work).
2. `Ti.App.Android.R` don't have any properties (except inherited from `Ti.Proxy`)

First problem could be solved in the generator solely. But the second couldn't, we need to change the docs.
Currently in docs there are 4 `R`s: 2 constants and 2 classes in `Ti.App.Android` and `Ti.Android` namespaces, but in the code there are only 3 `R`: 2 properties and single `RProxy`, so these changes reflect the actual state.
We can use `R` name for the common class (instead of `RProxy), naming does not really matter, but we can't use `Ti.Android.R` because it is also type of `Ti.App.Android.R` (anyway this common class/interface will not be exported from the module and only used internally as type for the `R` properties).

File: apidoc/Titanium/App/Android/Android.yml [hunk 1]
```diff
@@ -11,7 +11,26 @@ platforms: [android]
...
+        For example, if the file
+        `platform/android/res/drawable/icon.png` exists in the project, it can be accessed using its
+        integer id, as follows.
+
+            var resid = Titanium.App.Android.R.drawable.icon;
+
+        To access system-wide resources, rather than application resources, use <Titanium.Android.R>
+        instead.
+
+        Starting in Titanium SDK 7.3.0, this API exposes all Android related R classes
```
[the remark this answers is not in the document]
Comment (drauggres) [reply apidoc/Titanium/App/Android/Android.yml]: I removed `Starting in Titanium SDK 7.3.0 ...`, but `To access ..` should stay (I also added similar line to `Ti.Android.R` description).

File: apidoc/common/RProxy.yml [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
-    class, and should have the same syntax. For example, to retrieve the "OK" string in Android:
-    
+    These properties and sub-properties correspond directly to the properties of the the `android.R`
+    class, and should have the same syntax.
+    To retrieve the "OK" string from system resources in Android:
+
         Ti.Android.currentActivity.getString(Ti.Android.R.string.ok);
 
+    If the file  `platform/android/res/drawable/icon.png` exists in the project, it can be accessed
+    using its integer id, as follows:
```
[the remark this answers is not in the document]
Comment (drauggres) [reply apidoc/common/RProxy.yml]: fixed

File: apidoc/Titanium/App/Android/Android.yml [hunk 3]
```diff
@@ -11,7 +11,26 @@ platforms: [android]
...
-    type: Titanium.App.Android.R
+    description: |
+        The properties and sub-properties of this object relate directly to the native application
+        resources `R` class, and thus have the same syntax. For more information, refer to the
+        official documentation on the Android Developer website about
+        [accessing application resources](https://developer.android.com/guide/topics/resources/accessing-resources.html).
+
+        For example, if the file
+        `platform/android/res/drawable/icon.png` exists in the project, it can be accessed using its
+        integer id, as follows.
```
[the remark this answers is not in the document]
Comment (drauggres) [reply apidoc/Titanium/App/Android/Android.yml]: > Not sure why it mentions an integer id here, but resource name should be the correct phrasing.

fixed

> Also, please reformat the newlines on this paragraph so it uses up to 80 characters on each line.

Done.
P.S. It doesn't seem like we have 80 characters limit in the docs, most likely 100.

File: apidoc/Titanium/Android/Android.yml [hunk 4]
```diff
@@ -2148,7 +2148,7 @@ properties:
 
   - name: R
     summary: Accessor for Android system resources.
-    type: Titanium.Android.R
+    type: RProxy
```
[the remark this answers is not in the document]
Comment (drauggres) [reply apidoc/Titanium/Android/Android.yml]: Changed to `R`, explanation below in the comment.

File: apidoc/Titanium/App/Android/Android.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (janvennemann) [reply apidoc/Titanium/App/Android/Android.yml]: Yeah, we don't really have a character limit. The 80 characters are more of a guideline.

**Comment by janvennemann** [comment]
@drauggres We should be able to have `Titanium.Android.R` as both a value and a type, TypeScript supports that. Here's what i think needs to be done to solve this:

- Get rid of the additional R.yml under titanium/app/android (as you already did). Change the type of the `R` property in titanium/Android.yml to the existing `Titanium.Android.R`.
- Keep the R.yml under titanium/android with it's existing name `Titanium.Android.R`
- Change the TypeScript generator so it declares `Titanium.Android.R` as an interface instead of a namespace. Since TypeScript treats namespaces as values and interfaces as types, we can have both the `R` property on the `Titanium.Android` namespace, as well as the type definition `Titanium.Android.R` as an interface.

**Comment by drauggres** [comment]
>Keep the R.yml under titanium/android with it's existing name Titanium.Android.R

Then we'll have `Ti.Android.R` as type of `Ti.App.Android.R`. Don't you find it a bit confusing?

**Comment by janvennemann** [comment]
@drauggres No not really. As you said it is solely used internally to provide typing information. There is no way to create this type so users only access it through the properties `Ti.Android.R` and `Ti.App.Android.R`. With your updated summaries on the type and the properties, it should make it clear what's what.

Type:

> The Titanium binding of the native Android `R` class, giving access to Android system-wide resources **or** application resources.

And then the property definitions state:

Titanium.Android.R

> Accessor for Android **system resources**.

Titanium.App.Android.R

> The `R` namespace for **application resources**

**Comment by drauggres** [comment]
- `R` moved back to `Titanium.Android.R`.
- Type of `Ti.App.Android.R` changed to `Ti.Android.R`.

**Review by janvennemann — APPROVED** [review]

**Pull request merged**